src/main/java/org/elasticsearch/action/search/TransportSearchScrollAction.java

```diff
@@ -78,6 +78,19 @@ protected void doExecute(SearchScrollRequest request, ActionListener<SearchRespo
         }
     }
 
+    private void parseSource(SearchScrollRequest request) {
```

Comment:
Can you add a comment here that it is important to do the parsing here before we delegate to any of the actions? (otherwise we may copy unsafe references to source and we don't have the beforeLocalFork method infrastructure in search scroll)